datalad/core/local/resulthooks.py

```diff
@@ -0,0 +1,177 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Interface utility functions
+
+"""
+
+__docformat__ = 'restructuredtext'
+
+import logging
+import json
+
+lgr = logging.getLogger('datalad.core.local.hooks')
+
+
+def get_hooks_from_config(cfg):
+    """Parse out hook definitions given a ConfigManager instance
+
+    Returns
+    -------
+    dict
+      where keys are hook names/labels, and each value is a dict with
+      three keys: 'cmd' contains the name of the to-be-executed DataLad
+      command; 'args' has a JSON-encoded string with a dict of keyword
+      arguments for the command (format()-language based placeholders
+      can be present); 'match' holds a JSON-encoded string representing
+      a dict with key/value pairs that need to match a result in order
+      for a hook to be triggered.
+    """
+    hooks = {}
+    for h in (k for k in cfg.keys()
+              if k.startswith('datalad.result-hook.')
+              and k.endswith('.match')):
+        proc = cfg.get('{}.proc'.format(h[:-6]), None)
```

Comment:
Hmm, so `proc` is for "procedure".  While that could be a fairly general term, we already have procedures, so I think that's a bit confusing here given it's about calling any dataset method, not just `run_procedure`.  How about `s/.proc/.call/` ?

Reply:
Will do, but go one step further. See below...